### ISSUE TITLE
libpam

### ISSUE THREAD

**Issue by spelle** [titi38/libnavajo#6]
Hi Thierry,

I updated to Ubuntu 16.04. I had to reinstall everything. When compiling libnavajo, It made me think that maybe you should update README with a section like Pre-requisite for installing libpam0g-dev (ubuntu)

$ sudo apt install libpam0g-dev

Hth.

Cheers,

**Comment by titi38** [comment]
Hi Sébastien,
All PAM dependancies have been removed in the new version 1.2 (for reasons
of consistancy and portability)
You should switch to branch v1.2_devel (this version is almost finalized
and tested)
Cheers

2016-04-24 22:41 GMT+02:00 Sébastien PELLÉ notifications@github.com:

**Comment by spelle** [comment]
Hi Thierry,

Ok. I am already using the 1.2_devel. Just pulled it.

I removed the libpam0g-dev.

Still having the issue : 

```
[sherpa@sagarmatha libnavajo (v1.2_devel ✗)]$ make
Scanning dependencies of target navajo
[  4%] Building CXX object CMakeFiles/navajo.dir/src/LocalRepository.cc.o
[  8%] Building CXX object CMakeFiles/navajo.dir/src/LogRecorder.cc.o
[ 12%] Building CXX object CMakeFiles/navajo.dir/src/LogFile.cc.o
[ 16%] Building CXX object CMakeFiles/navajo.dir/src/LogSyslog.cc.o
[ 20%] Building CXX object CMakeFiles/navajo.dir/src/LogStdOutput.cc.o
[ 25%] Building CXX object CMakeFiles/navajo.dir/src/WebServer.cc.o
[ 29%] Building CXX object CMakeFiles/navajo.dir/src/WebSocketClient.cc.o
[ 33%] Building CXX object CMakeFiles/navajo.dir/src/MPFDParser/Parser.cc.o
[ 37%] Building CXX object CMakeFiles/navajo.dir/src/MPFDParser/Field.cc.o
[ 41%] Building CXX object CMakeFiles/navajo.dir/src/MPFDParser/Exception.cc.o
[ 45%] Linking CXX shared library lib/libnavajo.so
/usr/bin/ld: cannot find -lpam
collect2: error: ld returned 1 exit status
CMakeFiles/navajo.dir/build.make:331: recipe for target 'lib/libnavajo.so.1.2.0' failed
make[2]: *** [lib/libnavajo.so.1.2.0] Error 1
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/navajo.dir/all' failed
make[1]: *** [CMakeFiles/navajo.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2
```

I removed references to `pam`wihtin `CMakeLists.txt`.

```
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 058a481..ec8fb74 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -99,7 +99,7 @@ set_target_properties(navajoStatic PROPERTIES OUTPUT_NAME navajo)
 include_directories (${PROJECT_SOURCE_DIR}/include)

 target_link_libraries(navajo ${OPENSSL_LIBRARIES})
-target_link_libraries(navajo dl pam)
+target_link_libraries(navajo dl)
 target_link_libraries(navajo ${ZLIB_LIBRARIES})

 ############### install the library ###################
@@ -117,7 +117,7 @@ file( GLOB  navajoPrecompiler_source ${PROJECT_SOURCE_DIR}/src/navajoPrecompiler
 add_executable(navajoPrecompiler ${navajoPrecompiler_source})

 target_link_libraries(navajoPrecompiler ${OPENSSL_LIBRARIES})
-target_link_libraries(navajoPrecompiler dl pam)
+target_link_libraries(navajoPrecompiler dl)
 target_link_libraries(navajoPrecompiler ${ZLIB_LIBRARIES})

 install(TARGETS navajoPrecompiler DESTINATION bin COMPONENT headers)
```

Hope this helps.

Cheers,

**Comment by spelle** [comment]
I modified `dpkBuild.sh` : 

```
diff --git a/dpkgBuild.sh b/dpkgBuild.sh
index 72f7d6b..102474c 100755
--- a/dpkgBuild.sh
+++ b/dpkgBuild.sh
@@ -39,7 +39,7 @@ Section: Developpement
 Priority: optional
 Maintainer: Thierry DESCOMBES <thierry.descombes@gmail.com>
 Architecture: amd64 
-Depends: openssl, zlib1g-dev, libpam-dev
+Depends: openssl, zlib1g-dev
 Version: 1.2
 Description: an implementation of a complete HTTP(S) server, complete, fast and lightweight.
 EOF_CONTROL
```

Cheers.

**Comment by titi38** [comment]
Hi Sébastien,
I have done some minor changes (like the one you suggest ;-) and I have
merged the branch v1.2_devel a few hours ago.
The final v1.2 is now available in the master branch...
Cheers

2016-04-25 21:28 GMT+02:00 Sébastien PELLÉ notifications@github.com:
